src/wazuh_modules/task_manager/wm_task_manager.c

```diff
@@ -0,0 +1,310 @@
+                break;
+            }
+        } else {
+            error_code = INVALID_MESSAGE;
+            mterror(WM_TASK_MANAGER_LOGTAG, MOD_TASK_UNKNOWN_VALUE_ERROR, json_keys[MODULE], module, task);
+            break;
+        }
+    }
+
+    if (error_code) {
+        *response = wm_task_manager_build_response_error(error_code);
+    } else {
+        *response = cJSON_PrintUnformatted(response_array);
+    }
+
+    mtdebug1(WM_TASK_MANAGER_LOGTAG, MOD_TASK_RESPONSE_MESSAGE, *response);
+
+    cJSON_Delete(event_array);
+    cJSON_Delete(response_array);
+
+    return strlen(*response);
+}
+
+cJSON* wm_task_manager_parse_message(const char *msg) {
+    cJSON *event_array = NULL;
+    cJSON *task_object = NULL;
+    cJSON *module_json = NULL;
+    cJSON *command_json = NULL;
+    const char *error;
+    int task = 0;
+    int tasks = 0;
+
+    // Parsing event
+    if (event_array = cJSON_ParseWithOpts(msg, &error, 0), !event_array) {
+        mterror(WM_TASK_MANAGER_LOGTAG, MOD_TASK_PARSE_JSON_ERROR, msg);
+        return NULL;
+    }
+
+    // Getting array size
+    if (tasks = cJSON_GetArraySize(event_array), !tasks) {
+        mterror(WM_TASK_MANAGER_LOGTAG, MOD_TASK_PARSE_EMPTY_ERROR);
+        cJSON_Delete(event_array);
+        return NULL;
+    }
+
+    for (task = 0; task < tasks; ++task) {
+        // Getting task
+        task_object = cJSON_GetArrayItem(event_array, task);
+
+        // Detect module
+        if (module_json = cJSON_GetObjectItem(task_object, json_keys[MODULE]), !module_json) {
+            mterror(WM_TASK_MANAGER_LOGTAG, MOD_TASK_PARSE_KEY_ERROR, json_keys[MODULE], task);
+            cJSON_Delete(event_array);
+            return NULL;
+        }
+
+        // Detect command
+        if (command_json = cJSON_GetObjectItem(task_object, json_keys[COMMAND]), !command_json) {
+            mterror(WM_TASK_MANAGER_LOGTAG, MOD_TASK_PARSE_KEY_ERROR, json_keys[COMMAND], task);
+            cJSON_Delete(event_array);
+            return NULL;
+        }
+    }
+
+    return event_array;
+}
+
+cJSON* wm_task_manager_build_response_insert(int agent_id, int task_id) {
+    cJSON *response_json = cJSON_CreateObject();
+
+    cJSON_AddNumberToObject(response_json, json_keys[ERROR], SUCCESS);
+    cJSON_AddStringToObject(response_json, json_keys[ERROR_DATA], error_codes[SUCCESS]);
+    cJSON_AddNumberToObject(response_json, json_keys[AGENT_ID], agent_id);
+    cJSON_AddNumberToObject(response_json, json_keys[TASK_ID], task_id);
+
+    return response_json;
+}
+
+char* wm_task_manager_build_response_error(int error_code) {
+    cJSON *error_json = cJSON_CreateObject();
+    char *error = NULL;
+
+    cJSON_AddNumberToObject(error_json, json_keys[ERROR], error_code);
+    cJSON_AddStringToObject(error_json, json_keys[ERROR_DATA], error_codes[error_code]);
+
+    error = cJSON_PrintUnformatted(error_json);
+
+    cJSON_Delete(error_json);
+
+    return error;
+}
```

Comment:
This method and `wm_task_manager_build_response_insert` could be one, (or maybe one a smaller part of the other)

Reply:
Done!